hyperspy/drawing/markers.py

```diff
@@ -657,7 +657,7 @@ def get_current_kwargs(self, only_variable_length=False):
         """
         current_keys = {}
         if self._is_iterating:
-            indices = self._axes_manager.indices[::-1]
```

Comment:
This is currently causing the `from_signal` method to silently fail.  You need to reverse the indexes to follow the array--> plotting specification.